### PR TITLE
[WIP] Settle on `Swapchain` naming convention.

### PR DESCRIPTION
**Pull request by Binero**

### CODE REVIEW

File: src/backend/dx12ll/src/lib.rs [hunk 1]
```diff
@@ -361,14 +361,14 @@ impl core::Surface for Surface {
     }
 }
 
-pub struct SwapChain {
-    inner: ComPtr<winapi::IDXGISwapChain3>,
+pub struct Swapchain {
+    inner: ComPtr<winapi::IDXGISwapchain3>,
```
Comment (kvark) [inline src/backend/dx12ll/src/lib.rs]: how does this work?

File: src/window/dxgi/src/lib.rs [hunk 2]
```diff
@@ -228,14 +228,14 @@ pub struct Surface12 {
...
+    type Swapchain = Swapchain12;
 
     fn supports_queue(&self, _: &device_dx12::QueueFamily) -> bool { true }
-    fn build_swapchain<Q>(&mut self, config: core::SwapchainConfig, present_queue: &Q) -> SwapChain12
+    fn build_swapchain<Q>(&mut self, config: core::SwapchainConfig, present_queue: &Q) -> Swapchain12
         where Q: AsRef<device_dx12::CommandQueue>
     {
         use core::handle::Producer;
-        let mut swap_chain: *mut winapi::IDXGISwapChain1 = ptr::null_mut();
+        let mut swap_chain: *mut winapi::IDXGISwapchain1 = ptr::null_mut();
```
Comment (kvark) [inline src/window/dxgi/src/lib.rs]: same here

File: src/backend/dx12ll/src/lib.rs (hunk 1 above)
Comment (Binero) [inline src/backend/dx12ll/src/lib.rs]: It's hard to trace these down without a working Windows environment, so I opened the PR to see which ones need fixing.

**Comment by msiglreith** [comment]
Let's go with vulkan here imo.

**Comment by kvark** [comment]
We need to fix the AppVeyor script here to actually build dx12

**Comment by kvark** [comment]
:/

**Comment by notriddle** [comment]
There's an obnoxious backlog for OSX builds on Travis CI. https://www.traviscistatus.com/

**Comment by notriddle** [comment]
It was down when that second `bors r+` was run. https://bors.tech/pants-on-head-stupid,/downtime,/retrospective/2017/08/24/we-were-down/

Sorry about that. Took me *way too long* to figure out that it was down.

**Comment by notriddle** [comment]
@kvark You thumbed-up it. Are you going to re-r+ the PR?

**Pull request merged**